drivers/amplifiers/ada4250/ada4250.c

```diff
@@ -272,29 +314,62 @@ int32_t ada4250_init(struct ada4250_dev **device,
 	dev->offset_uv = init_param->offset_uv;
 	dev->avdd_v = init_param->avdd_v;
 
-	/* SPI */
-	ret = spi_init(&dev->spi_desc, init_param->spi_init);
-	if (ret != SUCCESS)
-		goto error_dev;
+	if(dev->device_id == ADA4250) {
+		/* SPI */
+		ret = spi_init(&dev->spi_desc, init_param->spi_init);
+		if (ret != SUCCESS)
+			goto error_dev;
 
-	ret = ada4250_read(dev, ADA4250_REG_CHIP_ID2, &data);
-	if (ret != SUCCESS)
-		goto error_spi;
+		ret = ada4250_read(dev, ADA4250_REG_CHIP_ID2, &data);
+		if (ret != SUCCESS)
+			goto error_spi;
 
-	chip_id = data;
+		chip_id = data;
 
-	ret = ada4250_read(dev, ADA4250_REG_CHIP_ID1, &data);
-	if (ret != SUCCESS)
-		goto error_spi;
+		ret = ada4250_read(dev, ADA4250_REG_CHIP_ID1, &data);
+		if (ret != SUCCESS)
+			goto error_spi;
 
-	chip_id = (chip_id << 8 | data);
+		chip_id = (chip_id << 8 | data);
 
-	if (chip_id != ADA4250_CHIP_ID)
-		goto error_spi;
+		if (chip_id != ADA4250_CHIP_ID)
+			goto error_spi;
 
-	ret = ada4250_soft_reset(dev);
-	if(ret != SUCCESS)
-		return FAILURE;
+		ret = ada4250_soft_reset(dev);
+		if(ret != SUCCESS)
+			return FAILURE;
+	} else {
+		ret = gpio_get_optional(&dev->gpio_g2, init_param->gpio_g2_param);
+		if (ret != SUCCESS)
+			goto error_dev;
+
+		ret = gpio_direction_output(dev->gpio_g2, GPIO_LOW);
+		if (ret != SUCCESS)
+			goto error_g2;
+
+		ret = gpio_get_optional(&dev->gpio_g1, init_param->gpio_g1_param);
+		if (ret != SUCCESS)
+			goto error_g2;
+
+		ret = gpio_direction_output(dev->gpio_g1, GPIO_LOW);
+		if (ret != SUCCESS)
+			goto error_g1;
+
+		ret = gpio_get_optional(&dev->gpio_g0, init_param->gpio_g0_param);
```

Comment:
If you use gpio_get_optional shouldn't later check every time if gpio_g0 is not NULL?

Reply:
Actually, `gpio_get` will fit better here, since the gpios are mandatory for ada4230.